projects/templates/projects/allocation.html

```diff
@@ -1,82 +1,62 @@
+<td style="padding-right:10px">
+  <small>{{ alloc.computeUsed }} / {{ alloc.computeAllocated }} SUs used ({{alloc.percentComputeUsed|floatformat:"0"}}%)</small>
+  <div class="progress">
+    <div class="progress-bar" role="progressbar" aria-valuenow="{{ alloc.computeUsed }}" aria-valuemin="0" aria-valuemax="{{ alloc.computeAllocated }}" style="width: {{ alloc.percentComputeUsed }}%"></div>
+  </div>
+</td>
+{% elif alloc.status == 'Rejected' %}
+<td>{{ alloc.decisionSummary }}</td>
+{% else %}
+<td></td>
+{% endif %}
+<td>
+  <div class="dropdown">
+    <button class="btn btn-secondary dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" {% if not can_manage_project %}disabled{%endif%}>
+      <i class="fa fa-gear text-info dropbtn"></i><span class="sr-only">Manage Allocation</span>
+    </button>
+    <div class="dropdown-menu dropdown-content" aria-labelledby="dropdownMenuButton">
+      {% if can_manage_project %}
+        {% if alloc.status == 'Approved' or alloc.status == 'Active' or alloc.status == 'Inactive' %}
+        <button class="btn btn-xs btn-block">
+          <a href="{% url 'projects:renew_allocation' project.id alloc.id %}">
+            Recharge/Extend Allocation
+          </a>
+        </button>
+        {% elif alloc.status == 'Rejected' %}
+        <button class="btn btn-xs btn-block">
+          <a href="{% url 'projects:renew_allocation' project.id alloc.id %}">
+            Resubmit Allocation
           </a>
+        </button>
         {% endif %}
-      </div>
-    </div>
-  {% endif %}
-
-  {% if alloc.status == 'Rejected' %}
-    <div class="row">
-      <div class="col-sm-3">
-        <h5>Date requested</h5>
-        {{ alloc.dateRequested|date:"d F, Y" }}
-      </div>
-      <div class="col-sm-3">
-        <h5>Date reviewed</h5>
-        {{ alloc.dateReviewed|date:"d F, Y" }}
-      </div>
-      <div class="col-sm-3">
-        <h5>Decision summary</h5>
-        {{ alloc.decisionSummary }}
-      </div>
-      <div class="col-sm-3">
-      {% if can_manage_project and not project.has_pending_allocations and not project.has_active_allocations and not project.has_approved_allocations %}
-        <h5>Actions</h5>
-        <a href="{% url 'projects:renew_allocation' project.id alloc.id %}"
-           title="">
-          <i class="fa fa-refresh"></i> Resubmit allocation
-        </a>
+      {% endif %}
+      <button class="btn btn-xs btn-block" {% if alloc.status != 'Approved' and alloc.status != 'Active' and alloc.status != 'Inactive' %}disabled{%endif%}>
+        {% if alloc.status != 'Approved' and alloc.status != 'Active' and alloc.status != 'Inactive' %}
+        View Charges
+        {% else %}
+        <a class="open-charge" href="{% url 'projects:view_charge' alloc.id %}" data-charge-popup-url="{% url 'projects:view_charge' alloc.id %}">View Charge</a>
```

Comment:
How does the charge get rendered inline? Is this something handled automagically by the modal JS helper?

Reply:
a function to open up the modal. https://github.com/ChameleonCloud/portal/blob/a3587decb2abd1c86f3b9e80948077b3e6827f6a/projects/static/projects/js/allocations.js#L12-L19. Don't know if answers your questions?